decred/tests/unit/dcr/test_dcrdata_unit.py

```diff
@@ -302,11 +266,97 @@ def receiver(obj):
     )
     # fmt: on
 
+    def test_misc(self, http_get_post, tmp_path):
+        preload_api_list(http_get_post)
+        http_get_post(f"{BASE_URL}api/block/best", dict(height=1))
+        ddb = DcrdataBlockchain(str(tmp_path / "test.db"), testnet, BASE_URL)
+        assert ddb.tipHeight == 1
+
+        # getAgendasInfo
+        http_get_post(f"{BASE_URL}api/stake/vote/info", AGENDAS_INFO_RAW)
+        agsinfo = ddb.getAgendasInfo()
+        assert isinstance(agsinfo, agenda.AgendasInfo)
+
+    def test_subscriptions(self, http_get_post, tmp_path):
+        # Exception in updateTip.
+        preload_api_list(http_get_post)
+        with pytest.raises(DecredError):
+            DcrdataBlockchain(str(tmp_path / "test.db"), testnet, BASE_URL)
+
+        # Successful creation.
+        preload_api_list(http_get_post)
+        http_get_post(f"{BASE_URL}api/block/best", dict(height=1))
+        ddb = DcrdataBlockchain(str(tmp_path / "test.db"), testnet, BASE_URL)
+
+        # Set the mock WebsocketClient.
+        ddb.dcrdata.ps = MockWebsocketClient()
+
+        # Receiver
+        receive_queue = []
+
+        def receiver(obj):
+            receive_queue.append(obj)
+
+        # subscribeBlocks
+        ddb.subscribeBlocks(receiver)
+        assert ddb.dcrdata.ps.sent[0]["message"]["message"] == "newblock"
+        ddb.dcrdata.ps.sent = []
+
+        # subscribeAddresses
+        with pytest.raises(DecredError):
+            ddb.subscribeAddresses([])
+        ddb.subscribeAddresses(["new_one"], receiver)
+        assert ddb.dcrdata.ps.sent[0]["message"]["message"] == "address:new_one"
+
+        # pubsubSignal
+        assert ddb.pubsubSignal("done") is None
+        assert ddb.pubsubSignal(dict(event="subscribeResp")) is None
+        assert ddb.pubsubSignal(dict(event="ping")) is None
+        assert ddb.pubsubSignal(dict(event="unknown")) is None
+        # pubsubSignal address
+        sig = dict(
+            event="address",
+            message=dict(address="the_address", transaction="transaction",),
+        )
+        ddb.pubsubSignal(sig)
+        assert receive_queue[0] == sig
+        receive_queue.clear()
+        # pubsubSignal newblock
+        sig = dict(event="newblock", message=dict(block=dict(height=1)))
+        ddb.pubsubSignal(sig)
+        assert receive_queue[0] == sig
+        receive_queue.clear()
+
     def test_utxos(self, http_get_post, tmp_path):
         preload_api_list(http_get_post)
-        http_get_post(f"{BASE_URL}api/block/best", 1)
+        http_get_post(f"{BASE_URL}api/block/best", dict(height=1))
```

Comment:
This was incorrect. The best block call returns a whole block, not directly the height.